src/elfpy/simulators.py

```diff
@@ -170,6 +173,63 @@ def override_variables(self, override_dict):
         if "init_share_price" in override_dict.keys():
             self.init_share_price = override_dict["init_share_price"]  # \mu variable
 
+    def setup_init_lp_agent(self):
+        """
+        Calculate the required deposit amounts and instantiate the LP agent
+
+        Arguments
+        ---------
+        None
+
+        Returns
+        -------
+        initial_lp : Agent
+            Agent class that will perform the lp initialization action
+        """
+        init_share_reserves, init_bond_reserves = price_utils.calc_liquidity(
+            target_liquidity=self.config.simulator.target_liquidity,
+            market_price=self.config.market.base_asset_price,
+            apr=self.config.simulator.init_pool_apy,
+            days_remaining=self.config.simulator.token_duration,
+            time_stretch=self.market.time_stretch_constant,
+            init_share_price=self.market.init_share_price,
+            share_price=self.market.init_share_price,
+        )[:2]
+        normalized_days_until_maturity = self.config.simulator.token_duration  # `t`; full duration
+        stretch_time_remaining = time_utils.stretch_time(
+            normalized_days_until_maturity, self.market.time_stretch_constant
+        )
+        output_with_fee = self.market.pricing_model.calc_out_given_in(
+            in_=init_bond_reserves,
+            share_reserves=init_share_reserves,
+            bond_reserves=0,
+            token_out="pt",
+            fee_percent=self.config.simulator.fee_percent,
+            time_remaining=stretch_time_remaining,
+            init_share_price=self.market.init_share_price,
+            share_price=self.market.init_share_price,
+        )[1]
+        amount_to_lp = init_share_reserves + output_with_fee
```

Comment:
the total amount to deposit initially?  we need extra because we are going to be doing a trade?